### PR TITLE
feat(Core/GameEvents)Add new command & load function to reload game event npc vendors

### LINKED ISSUE #20306: Feature: game_event_npc_vendor cannot be reloaded
### Describe your feature request or suggestion in detail

Make game_event_vendor reloadable same way npc_vendor can be reloaded

### Describe a possible solution to your feature or suggestion in detail

Command, game event vendor table, reload

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Exitare**
## Changes Proposed:
This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:
- Closes #20306 

## SOURCE:
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:
This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:
1. .event start 10 (Childrens week)
2. .npc add 29478
3. add a new entry in the game_event_npc_vendor database for guid 99369 with item 100 
4. reload the table & see the new item in the vendor list
5. .event stop 10, to see all items of the event being removed as you´d expect.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Grimdhex** [comment]
Hi, thank for the PR. Just, we will merged this PR: https://github.com/azerothcore/azerothcore-wotlk/pull/19858. For now I didn't take the time to review your PR but did you see a potential conflict/issue with arena vendor ?

Thanks for all your contributions, I plan to review it nearly ☺️

**Comment by Exitare** [comment]
Hi :) 
I skimmed over the linked PR. It doesn't seem to affect anything related to this PR. 
This PR only removes and adds items to vendors that are associated with an event, that is specified in the game_event table. 
I dont see any obvious conflicts.

File: src/server/game/Events/GameEventMgr.cpp [hunk 1]
```diff
@@ -236,6 +236,93 @@ void GameEventMgr::StopEvent(uint16 event_id, bool overwrite)
...
+            {
+                LOG_ERROR("sql.sql", "Invalid vendor item {} for entry {} during event {}, skipping.",
+                    newEntry.item, newEntry.entry, eventId);
+                continue;
+            }
+
+            // Add the item to the vendor if event is active
+            if (IsEventActive(eventId)) {
+                sObjectMgr->AddVendorItem(newEntry.entry, newEntry.item, newEntry.maxcount, newEntry.incrtime, newEntry.ExtendedCost, false);
+            }
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.cpp:314]: no need for curlies on one liners. Also curlies always start on a newline

File: src/server/game/Events/GameEventMgr.cpp [hunk 2]
```diff
@@ -236,6 +236,93 @@ void GameEventMgr::StopEvent(uint16 event_id, bool overwrite)
         sScriptMgr->OnGameEventStop(event_id);
 }
 
+void GameEventMgr::LoadNPCVendor()
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.cpp:239]: I would rename it to LoadEventVedors()

File: data/sql/updates/pending_db_world/rev_1734055142233360300.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+--
+DELETE FROM `command` where `name` = 'reload game_event_npc_vendor';
+INSERT INTO `command` (`name`, `security`, `help`) VALUES ('reload game_event_npc_vendor', '3', 'Syntax: .reload game_event_npc_vendor\r Reload game_event_npc_vendor table.');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1734055142233360300.sql:3]:
```suggestion
INSERT INTO `command` (`name`, `security`, `help`) VALUES ('reload game_event_npc_vendor', 3, 'Syntax: .reload game_event_npc_vendor\r Reload game_event_npc_vendor table.');
```

File: src/server/game/Events/GameEventMgr.cpp [hunk 4]
```diff
@@ -236,6 +236,92 @@ void GameEventMgr::StopEvent(uint16 event_id, bool overwrite)
...
+    WorldDatabasePreparedStatement* stmt = WorldDatabase.GetPreparedStatement(WORLD_SEL_GAME_EVENT_NPC_VENDOR);
+    PreparedQueryResult result = WorldDatabase.Query(stmt);
+
+    if (!result)
+    {
+        LOG_WARN("server.loading", ">> Loaded 0 Vendor Additions In Game Events. DB Table `game_event_npc_vendor` Is Empty.");
+        LOG_INFO("server.loading", " ");
+    }
+    else
+    {
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.cpp:251]:
```suggestion
    {
        LOG_WARN("server.loading", ">> Loaded 0 Vendor Additions In Game Events. DB Table `game_event_npc_vendor` Is Empty.");
        LOG_INFO("server.loading", " ");
        return;
    }
```

File: src/server/game/Events/GameEventMgr.cpp [hunk 5]
```diff
@@ -236,6 +236,92 @@ void GameEventMgr::StopEvent(uint16 event_id, bool overwrite)
...
+        std::unordered_set<uint8> processedEvents;
+
+        do
+        {
+            Field* fields = result->Fetch();
+            uint8 eventId = fields[0].Get<uint8>();
+
+            if (eventId >= mGameEventVendors.size())
+            {
+                LOG_ERROR("sql.sql", "`game_event_npc_vendor` game event id ({}) is out of range compared to max event id in `game_event`", eventId);
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.cpp:262]:
```suggestion
                LOG_ERROR("sql.sql", "Table `game_event_npc_vendor` has invalid eventEntry ({}) for GUID ({}), skipped.", eventId, guid.ToString());
```

File: src/server/game/Events/GameEventMgr.cpp [hunk 6]
```diff
@@ -236,6 +236,92 @@ void GameEventMgr::StopEvent(uint16 event_id, bool overwrite)
...
+
+            // Get creature entry
+            newEntry.entry = 0;
+            if (CreatureData const* data = sObjectMgr->GetCreatureData(guid))
+                newEntry.entry = data->id1;
+
+            // Validate vendor item
+            if (!sObjectMgr->IsVendorItemValid(newEntry.entry, newEntry.item, newEntry.maxcount, newEntry.incrtime, newEntry.ExtendedCost, nullptr, nullptr, event_npc_flag))
+            {
+                LOG_ERROR("sql.sql", "Invalid vendor item {} for entry {} during event {}, skipping.",
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.cpp:306]:
```suggestion
                LOG_ERROR("sql.sql", "Table `game_event_npc_vendor` has invalid item ({}) for guid ({}) for event ({}), skipped.",
```

**Comment by Exitare** [comment]
@Kitzunu  done :)

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**